src/Core/KeyManagement/Queries/IsV2EncryptionUserQuery.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Bit.Core.Entities;
+using Bit.Core.Enums;
+using Bit.Core.KeyManagement.Queries.Interfaces;
+using Bit.Core.KeyManagement.Repositories;
+using Bit.Core.KeyManagement.Utilities;
+
+namespace Bit.Core.KeyManagement.Queries;
+
+public class IsV2EncryptionUserQuery(IUserSignatureKeyPairRepository userSignatureKeyPairRepository)
+    : IIsV2EncryptionUserQuery
+{
+    public async Task<bool> Run(User user)
+    {
+        ArgumentNullException.ThrowIfNull(user);
+
+        var hasSignatureKeyPair = await userSignatureKeyPairRepository.GetByUserIdAsync(user.Id) != null;
+        var isPrivateKeyEncryptionV2 =
+            !string.IsNullOrWhiteSpace(user.PrivateKey) &&
+            EncryptionParsing.GetEncryptionType(user.PrivateKey) == EncryptionType.XChaCha20Poly1305_B64;
+
+        return hasSignatureKeyPair switch
```

Comment:
To be honest, this is nit-picky, but this would be more readable as an if/else. If this were rust matching this would be different but it took me a second to get this.
i.e 
```
if (isPrivateKeyEncryptionV2 && hasSignatureKeyPair) {

} else if (!isPrivateKeyEncryptionV2 && !hasSignatureKeyPair) {

} else {
  throw
}
```